<a href="https://colab.research.google.com/github/aksalas/Video-Game-Sales/blob/main/Video_Game_Sales.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
# Imports

# Pandas
import pandas as pd
# Numpy
import numpy as np
# MatPlotLib
import matplotlib.pyplot as plt

# Preprocessing
from sklearn.model_selection import train_test_split
from sklearn.compose import make_column_selector, ColumnTransformer
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import StandardScaler, OneHotEncoder
from sklearn.pipeline import make_pipeline

# Models
from sklearn.dummy import DummyRegressor
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import RandomForestRegressor
from sklearn.ensemble import BaggingRegressor
from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import GridSearchCV

# Regression Metrics
from sklearn.metrics import r2_score, mean_absolute_error, mean_squared_error

# Set the default transformation output to Pandas
from sklearn import set_config
set_config(transform_output='pandas')

from IPython.core.display import clear_output

In [3]:
df = pd.read_csv('https://docs.google.com/spreadsheets/d/e/2PACX-1vQVKSuvxFHIkz3xVkpLKQaqSc8bGafnJiA0f-PXmQeFLCjFWxX7YAM6wHapPVuiELaOaJKjE0D3AFlS/pub?output=csv')

In [4]:
df.head()

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

1. https://www.kaggle.com/datasets/gregorut/videogamesales?resource=download
2. Video game sales (in Millions) and their respective genres and developers
3. target = global sales
4. a row represents a product
5. Regression
6. 11 features
7. 16598 rows
8. I dont see any forseeable challenges with this dataset as of yet.

In [9]:
df.shape

(16598, 11)

In [6]:
df2 = pd.read_csv('https://docs.google.com/spreadsheets/d/e/2PACX-1vSH7pYkTNbwac9XrddWv3fa79YbUtMVcU6ufjlnTomi56n1x3HTVxAurCDi-h4HfR1yKuo3trCWDBIs/pub?output=csv')

In [7]:
df2.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

In [10]:
df2.shape

(48842, 15)

1. https://www.kaggle.com/datasets/wenruliu/adult-income-dataset
2. Income of adults alongside some of their demographic information
3. target = Income
4. a row represents a person
5. Regression
6. 16 features
7. 48842 rows
8. I see challenges in cleaning the data